src/processor/Package.ts

```diff
@@ -83,4 +87,30 @@ export class Package {
       pullAt(sd.rules, rulesToRemove);
     });
   }
+
+  private constructNamedExtensionContainsRules(): void {
+    [...this.profiles, ...this.extensions].forEach(sd => {
+      const rulesToRemove: number[] = [];
+      sd.rules.forEach(rule => {
+        if (rule instanceof ExportableContainsRule && rule.path.endsWith('extension')) {
+          rule.items.forEach(item => {
+            const onlyRuleIdx = sd.rules.findIndex(
+              other =>
+                other.path === `${rule.path}[${item.name}]` && other instanceof ExportableOnlyRule
+            );
+            const onlyRule = sd.rules[onlyRuleIdx] as ExportableOnlyRule;
+            if (
+              onlyRule &&
+              onlyRule.types.length == 1 &&
+              isUri(onlyRule.types[0].type.split('|')[0])
```

Comment:
Why does this apply only if the type is a URI?  I think that the way GoFSH works now, all types here would be URIs anyway, but... in theory, shouldn't `extension[birthSex] only uscore-birth-sex` work just as well as `extension[birthSex] only http://hl7.org/fhir/us/core/StructureDefinition/us-core-birthsex`? What is the use case we're trying to _avoid_ here?

Reply:
The issue I was hitting was actually one from USCore. Check out the us-core-race SD, and you will see that in the differential element for the `ombCategory` slice, the type is set to `Extension`:
```
"type": [{ "code": "Extension" }]
```
This is not really necessary from USCore, but it is there in the differential and I guess not invalid. My thinking at the time was that any input to GoFSH that should be turned into a "named" rule will be a URL, since it will be specified in `type.profile`. But if we later on optimize rules to point to other defined items by name rather than URL, and we do that optimization before this one, I think it could be a problem.

I could just explicitly rule out `Extension` I think? At the time that felt wrong, but now that I think about it, if that value is anything other than `Extension`, then I think it should be fair to assume that it is referring to some profile on `Extension`. Does that make sense to you?

Reply:
Thanks for explaining.  Yeah, that's weird -- but I agree, special-casing `Extension` seems like the best way to go.  And then you also don't need the `isValidUri` function anymore.  I'd add a comment, however, explaining why we're special-casing `Extension` (e.g., we've seen it in US Core).

Reply:
Ok @cmoesel, 265f269 explicitly rules out `Extension`, and we no longer use the `isUri` function.